docs/tutorial/part06.rst

```diff
@@ -2,5 +2,648 @@
+                'node_id': self.node_id,
+                'status': self.status,
+            },
+        }
+
+Officer::
+
+    @property
+    def serialize(self):
+        return {
+            'node_properties': {
+                'sourceID': self.sourceID,
+                'name': self.name,
+                'country_codes': self.country_codes,
+                'valid_until': self.valid_until,
+                'countries': self.countries,
+                'node_id': self.node_id,
+            },
+        }
+
+Other::
+
+    @property
+    def serialize(self):
+        return{
+            'node_properties': {
+                'sourceID': self.sourceID,
+                'name': self.name,
+                'countries': self.countries,
+                'valid_until': self.valid_until,
+                'node_id': self.node_id,
+            },
+        }
+
+Call the serialize methods on the helpers
+=========================================
+
+Now instead of your ``fetch_nodes`` function on your ``fetch_api/utils.py`` looking like this::
+
+    def fetch_nodes(fetch_info):
+        node_type       = fetch_info['node_type']
+        search_word     = fetch_info['name']
+        country         = fetch_info['country']
+        limit           = fetch_info['limit']
+        skip            = ((fetch_info['skip'] - 1) * limit)
+        jurisdiction    = fetch_info['jurisdiction']
+        node_set        = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, jurisdiction)
+        node_set.limit  = limit
+        node_set.skip   = skip
+        fetched_nodes   = node_set.all()
+
+        return fetched_nodes
+
+Change the return statement to::
+
+        return [node.serialize for node in fetched_nodes]
+
+That's just going to create a new array of dictionaries with the values of the serialized nodes.
+
+And your ``fetch_node_details`` function looking like this::
+
+    def fetch_node_details(node_info):
+        node_type       = node_info['node_type']
+        node_id         = node_info['node_id']
+        node            = MODEL_ENTITIES[node_type].nodes.get(node_id=node_id)
+
+        return node
+
+Should look like this::
+
+    def fetch_node_details(node_info):
+        node_type       = node_info['node_type']
+        node_id         = node_info['node_id']
+        node            = MODEL_ENTITIES[node_type].nodes.get(node_id=node_id)
+        node_details    = node.serialize
+
+        return node_details
+
+Basically doing the same as before but for a single node instead of a set.
+
+Create the serialize relationships methods
+==========================================
+
+We are not serializing the relationships along with the properties because:
+
+* That would create a loophole between nodes
+* You don't always need the relationships
+
+So, with this said, this is how we are gonna serialize the relationships, add the following to the
+file ``fetch_api/models/entity.py``::
+
+    from .nodeutils import NodeUtils

```

Comment:
Before this part at the tutorial, there's not a definition or process to create a module named nodeutils. It should be added.